sdk/rust/reference.mdx

```diff
@@ -1,4 +1,275 @@
+  .await.unwrap();
+
+let conn = db.connect().unwrap();
+```
+
+## Encryption
+
+To enable encryption on a SQLite file (`new_local` or `new_remote_replica`), pass the `encryption_config`:
+
+```rust
+use libsql::Builder;
+use bytes::Bytes;
+
+let url = env::var("LIBSQL_URL").expect("LIBSQL_URL must be set");
+let token = env::var("LIBSQL_AUTH_TOKEN").unwrap_or_default();
+
+let cipher = Cipher::YourChosenCipher;
+let encryption_key_bytes = Bytes::from("your_secure_encryption_key_here");
+
+let encryption_config = EncryptionConfig {
+    cipher,
+    encryption_key: encryption_key_bytes,
+};
+
+let mut db = Builder::new_remote_replica("local.db", &url, &token)
+  .encryption_config(encryption_config) // Apply encryption configuration
+  .build()
+  .await.unwrap();
+
+let conn = db.connect().unwrap();
+```
+
+<Info>
+
+Encrypted databases appear as raw data and cannot be read as standard SQLite databases. You must use the libSQL client for any operations — [learn more](/libsql#encryption-at-rest).
+
+</Info>
+
+## Simple query
+
+You can pass a string to `execute()` to invoke a SQL statement, as well as optional arguments:
+
+<CodeGroup>
+
+```rust Query
+conn.execute("SELECT * FROM users").await?;
+```
+
+```rust Arguments
+conn.execute("SELECT * FROM users WHERE id = ?1", [1]).await?;
+```
+
+</CodeGroup>
+
+## Prepared Statements
+
+You can prepare a cached statement using `prepare()` and then execute it with `query()`:
+
+```rust
+let stmt = db_conn.prepare("SELECT * FROM users").await?;
+
+db_conn.query(&stmt, &[&1]).await?;
+```
+
+## Placeholders
+
+libSQL supports the use of positional and named placeholders within SQL statements:
+
+<CodeGroup>
+
+```rust Positional
+conn.execute("SELECT * FROM users WHERE id = ?1", params![1]).await?;
+```
+
+```rust Named
+conn.execute("INSERT INTO users (name) VALUES (:name)", libsql::named_params! { ":name": "Iku" }).await?;
+```
+
+</CodeGroup>
+
+## Deserialization
+
+You can use the `de::from_row` function to deserialize a row into a struct:
+
+```rust
+use libsql::{de, Builder};
+
+let mut stmt = conn
+  .prepare("SELECT * FROM users WHERE id = ?1")
+  .await
+  .unwrap();
+let row = stmt
+  .query([1])
+  .await
+  .unwrap()
+  .next()
+  .await
+  .unwrap()
+  .unwrap();
+
+#[derive(Debug, serde::Deserialize)]
+#[allow(dead_code)]
+struct User {
+  name: String,
+  age: i64,
+  vision: f64,
+  avatar: Vec<u8>,
+}
+
+let user = de::from_row::<User>(&row).unwrap();
+```
+
+## Batch Transactions
+
+A batch consists of multiple SQL statements executed sequentially within an implicit transaction. The backend handles the transaction: success commits all changes, while any failure results in a full rollback with no modifications.
+
+```rust
+conn.execute_batch(r#"
+  CREATE TABLE IF NOT EXISTS users (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL
+  );
+
+  INSERT INTO users (name) VALUES ('Alice');
+  INSERT INTO users (name) VALUES ('Bob');
+"#).await?;
+```
+
+## Interactive Transactions
```

Comment:
would be good to mention for embedded replica usage we also support a `BEGIN READ ONLY` transaction type that will not allow writes and will forward all the queries to the local db instead of the remote one. Any transaction that is a write transaction will all get fowarded to the primary remote db.